Fix vt.Terminal failing test: test_log_sanitize

frebib opened the pull request:
Fixes failing test added in a09b4f445052be66f0ac53fd01fa02bfa5b82ea6

We can't assume tests are run at debug level, so this ensures the test
passes regardless of what logging level is currently set by capturing
the output in caplog at DEBUG which stream_stdout/stream_stderr uses by
default.

Signed-off-by: Joe Groocock <jgroocock@cloudflare.com>

---

This fails without this patch for us:
```python
tests/pytests/unit/utils/test_vt.py:30 (test_log_sanitize[echo])
****** !=
test_cmd = 'echo'
caplog = <_pytest.logging.LogCaptureFixture object at 0x7fc94f04d0d0>
    @pytest.mark.parametrize("test_cmd", ["echo", "ls"])
    @pytest.mark.skip_on_windows()
    def test_log_sanitize(test_cmd, caplog):
        """
        test when log_sanitize is passed in
        we do not see the password in either
        standard out or standard error logs
        """
        password = "123456"
        cmd = [test_cmd, password]
        term = vt.Terminal(
            cmd,
            log_stdout=True,
            log_stderr=True,
            log_sanitize=password,
            stream_stdout=False,
            stream_stderr=False,
        )
        ret = term.recv()
        assert password not in caplog.text
>       assert "******" in caplog.text
E       AssertionError: assert '******' in ''
E        +  where '' = <_pytest.logging.LogCaptureFixture object at 0x7fc94f04d0d0>.text
salt/tests/pytests/unit/utils/test_vt.py:51: AssertionError
```

@Ch3LL

frebib commented:
Not sure if I should target this PR at `master` or `freeze`. I'm happy to do either/both- please advise

Ch3LL commented:
@frebib thanks for this. The `master` branch works. For some reason the github action jobs are not running for you. Can you force a rebase and push to try to force them to run again.

Ch3LL commented:
Actually, I just realized all new submitted PRs are not running github actions and github status is showing issues with github actions so we will have to wait until those are resolved.

Ch3LL commented:
k, github actions should be working now, can you do a force push to re-start the github actions?

frebib committed:
Fix vt.Terminal failing test: test_log_sanitize

Fixes failing test added in a09b4f445052be66f0ac53fd01fa02bfa5b82ea6

We can't assume tests are run at debug level, so this ensures the test
passes regardless of what logging level is currently set by capturing
the output in caplog at DEBUG which stream_stdout/stream_stderr uses by
default.

Signed-off-by: Joe Groocock <jgroocock@cloudflare.com>

frebib commented:
> The `master` branch works

I've force-pushed. This is currently targetting `freeze`. Do you want me to re-target this to `master`?

frebib commented:
Never mind, apparently this test doesn't exist on `master` yet. This'll conflict with https://github.com/saltstack/salt/pull/62810 so if that's merged first, I'll have to rebase

Ch3LL commented:
k, I would prefer to get the merge forward in first since those can be difficult to merge in with the amount of changes. I'll label this 3006 so we do not lose track of this change.

frebib commented:
Sounds good. I'll rebase this as soon as the merge-formard is merged in

garethgreenaway committed:
Update test_vt.py

Pre-commit changes manually.

frebib commented on tests/pytests/unit/utils/test_vt.py:
```diff
@@ -3,7 +3,6 @@
 import signal
 
 import pytest
-
```
I'm reasonably sure `black` added this. Why are you removing it?

frebib replied:
Actually maybe it was `isort`

garethgreenaway replied:
It was failing the pre-commit check.

frebib commented:
This commit never made it into master? Can we merge freeze into master @Ch3LL @garethgreenaway ?

Ch3LL commented:
This was merged in a while ago, we no longer merge forward from freeze today. Would you be willing to cherry-pick it into master? If not I certainly can.

frebib commented:
I don't mind cherry-picking. My concern is that there may be others in this state. Maybe a merge freeze->master would be more suitable?

Ch3LL commented:
I know that we have added changes that are specific to this branch for tests and github pipelines. We did that knowing we are no longer merging forward this branch and its only receiving security updates. I would prefer to not do a merge forward for this reason.
